nsxt/resource_nsxt_policy_constraint_test.go

```diff
@@ -0,0 +1,266 @@
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtPolicyConstraintCheckDestroy(state, name)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNsxtPolicyConstraintMinimalistic(),
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNsxtPolicyConstraintExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy Constraint resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("Policy Constraint resource ID not set in resources")
+		}
+
+		exists, err := resourceNsxtPolicyConstraintExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("Policy Constraint %s does not exist", resourceID)
+		}
+
+		return nil
+	}
+}
+
+func testAccNsxtPolicyConstraintCheckDestroy(state *terraform.State, displayName string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_policy_constraint" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		exists, err := resourceNsxtPolicyConstraintExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err == nil {
+			return err
+		}
+
+		if exists {
+			return fmt.Errorf("Policy Constraint %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtPolicyConstraintTemplate(createFlow bool) string {
+	var attrMap map[string]string
+	if createFlow {
+		attrMap = accTestPolicyConstraintCreateAttributes
+	} else {
+		attrMap = accTestPolicyConstraintUpdateAttributes
+	}
+	return testAccNsxtPolicyProjectTemplate900(true, true, false, false) + fmt.Sprintf(`
+resource "nsxt_policy_constraint" "test" {
+  display_name = "%s"
+  description  = "%s"
+  message = "%s"
+
+  target {
+    path_prefix = "${nsxt_policy_project.test.path}/"
+  }
+
+  instance_count {
+    count                = "%s"
+    target_resource_type = "%s"
+  }
+}`, attrMap["display_name"], attrMap["description"], attrMap["message"], attrMap["count"], attrMap["target_resource_type"])
+}
+
+func testAccNsxtPolicyConstraintVpcTemplate(createFlow bool) string {
+	var attrMap map[string]string
+	if createFlow {
+		attrMap = accTestPolicyConstraintCreateAttributes
+	} else {
+		attrMap = accTestPolicyConstraintUpdateAttributes
+	}
+	return fmt.Sprintf(`
+data "nsxt_vpc" "test" {
+  %s
+  id = "%s"
+}
+
+resource "nsxt_policy_constraint" "test" {
+  %s
+  display_name = "%s"
+  description  = "%s"
+  message = "%s"
+
+  target {
+    path_prefix = "${data.nsxt_vpc.test.path}/"
```

Comment:
This is a bit ugly - if users will have to manipulate paths in that manner. Can we do that automatically if not suffixed by `/`?

Reply:
Good point, done